### PR TITLE
add api.md of v1

### LINKED ISSUE #178: A new API doc of v1 is needed.
As we are going to refactor cyclone,  we need to make a new API document.

### PR DESCRIPTION
**Pull request by zz-bmj**
## Fixes #178 

**Changes**
   
  - [x] add api.md of v1

### CODE REVIEW

File: docs/api/v1/api.md [hunk 1]
```diff
@@ -0,0 +1,1099 @@
...
+| DELETE | /projects/{project}/pipelines/{pipeline}/records/{recordId} |     |     | Delete a pipeline record | WIP, [link](#delete-a-pipeline-record) |
+| GET | /projects/{project}/pipelines/{pipeline}/records/{recordId}/logs |     | string | Get log of a pipeline record | WIP, [link](#get-log-of-a-pipeline-record) |
+
+## API Common
+
+### Prefix
+
+Add `/api/v1/` ahead, complete path: `/api/v1/` + `SubPath`
+
+### Path Parameter Explaination
```
Comment (supereagle) [inline docs/api/v1/api.md]: s/Explaination/Explanation/

File: docs/api/v1/api.md [hunk 2]
```diff
@@ -0,0 +1,1099 @@
...
+#### Request
+
+Http method: `POST`
+
+URL: `/api/v1/projects`
+
+Header:
+
+```
+X-Tenant: any
```
Comment (supereagle) [inline docs/api/v1/api.md]: `X-Tenant` header is not needed. There are same errors in other places.

File: docs/api/v1/api.md [hunk 3]
```diff
@@ -0,0 +1,1099 @@
...
+  "Owner": "string",                         // owner of the project
+  "UpdatedTime": "2017-08-23T09:44:08.653Z"  // updated time of the project
+}
+```
+
+Project is responsible for managing a set of applications, in this system, mainly managing a number of pipelining corresponding to this set of applications.
+
+### Add a project
+
+create a new project.
```
Comment (supereagle) [inline docs/api/v1/api.md]: The first letter should be uppercase. There are same errors in other places.

File: docs/api/v1/api.md [hunk 4]
```diff
@@ -0,0 +1,1099 @@
...
+### ProjectDataStructure
+
+```json
+{
+  "CreatedTime": "2017-08-23T09:44:08.653Z", // created time of the project
+  "Description": "string",                   // name of the project, should be unique
+  "ID": "string",                            // id of the project
+  "Name": "string",                          // name of the project, should be unique
+  "Owner": "string",                         // owner of the project
+  "UpdatedTime": "2017-08-23T09:44:08.653Z"  // updated time of the project
```
Comment (supereagle) [inline docs/api/v1/api.md]: 1. `CreatedTime` -> `createTime`, `UpdatedTime` -> `updateTime`
2. The first letter of the fields should be lowercase.
3. Will be better to simply order them.

File: docs/api/v1/api.md [hunk 5]
```diff
@@ -0,0 +1,1099 @@
...
+}
+```
+
+Project is responsible for managing a set of applications, in this system, mainly managing a number of pipelining corresponding to this set of applications.
```
Comment (supereagle) [inline docs/api/v1/api.md]: s/pipelining/pipelines/

File: docs/api/v1/api.md [hunk 6]
```diff
@@ -0,0 +1,1099 @@
...
+X-Tenant: any
+```
+
+Body:
+
+```
+{
+    "Description": "",                   // string, optional
+    "Name": "",                          // string, required
+    "Owner": ""                          // string, required
```
Comment (supereagle) [inline docs/api/v1/api.md]: The first letter of the fields should be lowercase. There are same errors in other places.

File: docs/api/v1/api.md [hunk 7]
```diff
@@ -0,0 +1,1099 @@
...
+  "Name": "string",
+  "Notification": {
+    "EmailNotification": {
+      "Emails": [
+        "string"
+      ]
+    },
+    "NotificationPolicy": "string"
+  },
+  "Owner": "string",
```
Comment (supereagle) [inline docs/api/v1/api.md]: Put these simple fields in the front of the data structure.

**Comment by zz-bmj** [comment]
@supereagle PTAL

File: docs/api/v1/api.md [hunk 8]
```diff
@@ -0,0 +1,833 @@
...
+#### Request
+
+Http method: `POST`
+
+URL: `/api/v1/projects/{project}/pipelines`
+
+Body:
+
+```json
+{
```
Comment (supereagle) [inline docs/api/v1/api.md]: Just put `PipelineDataStructure` here like the update API, instead of example data.

Comment (zz-bmj) [reply docs/api/v1/api.md]: done

**Comment by supereagle** [comment]
LGTM

**Pull request merged**